app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -436,11 +443,12 @@ private void hideInfoLayoutArcPointer() {
 
     @Override
     public void onDestroy() {
-        if (!batterySaver) {
+        super.onDestroy();
+
+        if (!batterySaver && timer != null) {
             timer.cancel();
         }
 
-        super.onDestroy();
```

Comment:
IIRC super.onDestroy had to be placed after `timer.cancel()`—and ideally it'd go at the end. I'd leave out unrelated changes.